src/thunderbots/software/ai/world/robot.cpp

```diff
@@ -38,7 +38,7 @@ void Robot::updateState(const Robot &new_robot_data)
             "Error: Robot updated using a robot with a mismatched id");
     }
 
-    for (int i = 0; i < new_robot_data.positions_.size(); i++)
+    for (int i = new_robot_data.positions_.size() - 1; i >= 0; i--)
```

Comment:
The issue here is the top item of the queue is the most recent, and this code places the newest item at the back of the new queue.